src-electron/rest/endpoint.js

```diff
@@ -70,26 +71,48 @@ function httpDeleteEndpointType(db) {
  */
 function httpPostEndpoint(db) {
   return async (request, response) => {
-    let { endpointId, networkId, profileId, endpointType, deviceIdentifier } =
-      request.body
+    let {
+      endpointId,
+      networkId,
+      profileId,
+      endpointType,
+      deviceIdentifier,
+      parentEndpointIdentifier,
+    } = request.body
     let sessionId = request.zapSessionId
+    let parentEndpointRef = await queryEndpoint.getParentEndpointRef(
+      db,
+      parentEndpointIdentifier,
+      sessionId
+    )
+    if (parentEndpointRef == null && parentEndpointIdentifier != null) {
+      parentEndpointIdentifier = null
+      notification.setNotification(
+        db,
+        'ERROR',
+        'Could not set Parent Endpoint because no Endpoint was found',
+        sessionId,
+        1,
+        1
+      )
+    }
     let newId = await queryEndpoint.insertEndpoint(
       db,
       sessionId,
       endpointId,
       endpointType,
       networkId,
       profileId,
-      deviceIdentifier
```

Comment:
This does not look right. Why is device identifier removed here?

Reply:
its not used by insertEndpoint because insertEndpoint does a LEFT JOIN from another table

Reply:
ok I see that `deviceIdentifier` is not used. Better to fix all the calls to `insertEndpoint` and make sure that only the following arguments are passed to it everywhere in the code.
       db,
       sessionId,
       endpointId,
       endpointType,
       networkId,
       profileId,
       parentEndpointRef